File: java-advent-of-code/src/test/java/advent/twenty_eighteen/Day23ExperimentalEmergencyTeleportationTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day23ExperimentalEmergencyTeleportationTest {
    @Test
    public void checkReal() {
        Day23ExperimentalEmergencyTeleportation cc = new Day23ExperimentalEmergencyTeleportation("/twenty_eighteen/Day23-ExperimentalEmergencyTeleportation-input.txt");

        assertEquals(674, cc.getPart1Answer().intValue());
        assertEquals(129444177, cc.getPart2Answer().longValue());
    }

    @Test
    public void checkExample1() {
        Day23ExperimentalEmergencyTeleportation cc = new Day23ExperimentalEmergencyTeleportation("/twenty_eighteen/Day23-ExperimentalEmergencyTeleportation-example#1.txt");

        assertEquals(7, cc.getPart1Answer().intValue());
        assertEquals(4, cc.getPart2Answer().longValue());
    }

    @Test
    public void checkExample2() {
        Day23ExperimentalEmergencyTeleportation cc = new Day23ExperimentalEmergencyTeleportation("/twenty_eighteen/Day23-ExperimentalEmergencyTeleportation-example#2.txt");

        assertEquals(6, cc.getPart1Answer().intValue());
        assertEquals(36, cc.getPart2Answer().longValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day23ExperimentalEmergencyTeleportationTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day23ExperimentalEmergencyTeleportationTest  extends FlatSpec with Matchers {
  "2018 Day 23 - Input" should "calculate answers" in {
    val dp = new Day23ExperimentalEmergencyTeleportation("/twenty_eighteen/Day23-ExperimentalEmergencyTeleportation-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day23ExperimentalEmergencyTeleportation("/twenty_eighteen/Day23-ExperimentalEmergencyTeleportation-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
